tests/util.py

```diff
@@ -70,12 +71,46 @@ def assert_format(
     safety guards so they don't just crash with a SyntaxError. Please note this is
     separate from TargetVerson Mode configuration.
     """
+    _assert_format_inner(
+        source, expected, mode, fast=fast, minimum_version=minimum_version
+    )
+
+    # For both preview and non-preview tests, ensure that Black doesn't crash on
+    # this code, but don't pass "expected" because the precise output may differ.
+    _assert_format_inner(
+        source,
+        None,
+        replace(mode, preview=not mode.preview),
+        fast=fast,
+        minimum_version=minimum_version,
+    )
+    # Similarly, setting line length to 1 is a good way to catch
+    # stability bugs. But only in non-preview mode because preview mode
+    # currently has a lot of line length 1 bugs.
+    _assert_format_inner(
+        source,
+        None,
+        replace(mode, preview=False, line_length=1),
+        fast=fast,
+        minimum_version=minimum_version,
+    )
+
+
+def _assert_format_inner(
+    source: str,
+    expected: Optional[str] = None,
+    mode: black.Mode = DEFAULT_MODE,
+    *,
+    fast: bool = False,
+    minimum_version: Optional[Tuple[int, int]] = None,
+) -> None:
     actual = black.format_str(source, mode=mode)
-    _assert_format_equal(expected, actual)
+    if expected is not None:
+        _assert_format_equal(expected, actual)
     # It's not useful to run safety checks if we're expecting no changes anyway. The
     # assertion right above will raise if reality does actually make changes. This just
     # avoids wasted CPU cycles.
```

Comment:
```suggestion
```
Unfortunately this optimization is now gone and I can't find a way to bring it back :(

Reply:
Not sure why? The optimization still applies.

Reply:
Ah, you are right. My bad.